### PR TITLE
Phcimg 927 dotty get fixes and handle dicom series description list

### PR DESCRIPTION
**Pull request by jiavila**
1. Fixes to dotty_dict.Dotty.get(), ensure returns None when indexing a nested list. 
- Note: Once dotty_dict PyPi is updated to include this fix, can remove this commit (NOT the PR!)

2. Fixes to dicom tag SeriesDescription. Should return a string, but handle case when a list is 
returned.
- Throws an error if anything other than a list or string is returned (lists get converted to strings)
- Added doctest tests in new common_utils method 'get_str_tag_from_dicom_header_dict()

### CODE REVIEW

**Comment by jiavila** [comment]
@NPann Passed sbx tests, ready for review

File: PT_classifier.py [hunk 1]
```diff
@@ -609,7 +638,8 @@ def classify_PT(df, dcm_metadata, acquisition):
     '''
     log.info("Determining PT Classification...")
     header_dicom = dcm_metadata['info']['header']['dicom']
-    series_description = header_dicom.get('SeriesDescription') or ''
+    series_description = common_utils.get_str_tag_from_dicom_header_dict(
+        header_dicom, tag_name='SeriesDescription')
```
Comment (NPann) [inline PT_classifier.py:642]: This issue has been addressed more generally in https://github.com/flywheel-apps/GRP-3/pull/39 so I am hesitant to have it patch in GRP-3B as well. What about removing here and releasing GRP-3 2.7.2 (which will fix the issue your reported as well) and GRP-3B together?

File: PT_classifier.py [hunk 2]
```diff
@@ -14,6 +14,33 @@
...
+            :param Any default: Default value if deep key does not exist
+            :return: Any or default value
+            """
+    try:
+        return self.__getitem__(key)
+    except (KeyError, IndexError):
+        return default
+
+
+Dotty.get = patch_get
```
Comment (NPann) [inline PT_classifier.py:37]: Given @kalebdfischer PR went through I'll be in favor of referencing the commit directly in the requirement to have a single line change. Adding the following to requirements.txt would do:
`-e git@github.com:pawelzny/dotty_dict.git@85be92023880b656904145ca077d015c1d6e327d#egg=dotty_dict`

And when they release to pipy we can just move the pointer

File: PT_classifier.py (hunk 1 above)
Comment (jiavila) [reply PT_classifier.py:642]: I see. Do you think there'll ever be a case where a string is expected but we get a list instead? If we're always going to take care of this problem in GRP-3B, then there's no need for this.

File: PT_classifier.py (hunk 2 above)
Comment (jiavila) [reply PT_classifier.py:37]: Sounds good. I'll go ahead and remove this and only change the requirements.txt file.

File: PT_classifier.py (hunk 1 above)
Comment (NPann) [reply PT_classifier.py:642]: For public tag GRP-3 will take care of the string/list discrepency. I guess the issue could still happen for private tags.

**Comment by jiavila** [comment]
@NPann OK, this updated PR passed us-sbx. I ran it with the udated GRP3 2.7.2 and no longer getting an issue with getting a list from dicom 'SeriesDescription' tag.

File: PT_classifier.py (hunk 2 above)
Comment (NPann) [reply PT_classifier.py:37]: Decided to wait for pip. the base docker image does not have git install.

**Pull request merged**